src/newgrf.cpp

```diff
@@ -2103,7 +2103,7 @@ static ChangeInfoResult CanalChangeInfo(uint id, int numinfo, int prop, ByteRead
 	ChangeInfoResult ret = CIR_SUCCESS;
 
 	if (id + numinfo > CF_END) {
-		grfmsg(1, "CanalChangeInfo: Canal feature %u is invalid, max %u, ignoring", id + numinfo, CF_END);
+		grfmsg(1, "CanalChangeInfo: Canal feature 0x%02X is invalid, max %u, ignoring", id + numinfo, CF_END);
```

Comment:
Should be able to use `%#02X` to get the 0x prefix

Reply:
%#02X will output 0XAA instead of 0xAA. But actually this one isn't a NewGRF feature, so...

Reply:
mm, documentation seems to suggest `0X` or `0x`, and being consistent is probably better